src/Responses/Threads/Messages/Delta/ThreadMessageDeltaResponse.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenAI\Responses\Threads\Messages\Delta;
+
+use OpenAI\Contracts\ResponseContract;
+use OpenAI\Responses\Concerns\ArrayAccessible;
+use OpenAI\Testing\Responses\Concerns\Fakeable;
+
+/**
+ * @implements ResponseContract<array{id: string, object: string, created_at: int, thread_id: string, role: string, content: array<int, array{type: string, image_file: array{file_id: string}}|array{type: string, text: array{value: string, annotations: array<int, array{type: string, text: string, file_citation: array{file_id: string, quote: string}, start_index: int, end_index: int}|array{type: string, text: string, file_path: array{file_id: string}, start_index: int, end_index: int}>}}>, assistant_id: ?string, run_id: ?string, file_ids: array<int, string>, metadata: array<string, string>}>
+ */
+final class ThreadMessageDeltaResponse implements ResponseContract
+{
+    /**
+     * @use ArrayAccessible<array{id: string, object: string, created_at: int, thread_id: string, role: string, content: array<int, array{type: string, image_file: array{file_id: string}}|array{type: string, text: array{value: string, annotations: array<int, array{type: string, text: string, file_citation: array{file_id: string, quote: string}, start_index: int, end_index: int}|array{type: string, text: string, file_path: array{file_id: string}, start_index: int, end_index: int}>}}>, assistant_id: ?string, run_id: ?string, file_ids: array<int, string>, metadata: array<string, string>}>
+     */
+    use ArrayAccessible;
+
+    use Fakeable;
+
+    private function __construct(
+        public string $id,
+        public string $object,
+        public ThreadMessageDeltaObject $delta,
+    ) {
+    }
+
+    /**
+     * Acts as static factory, and returns a new Response instance.
+     *
+     * @param  array{id: string, object: string, delta: array{role: string, content: array<int, array{type: 'image_file', image_file: array{file_id: string}}|array{type: 'text', text: array{value: string, annotations: array<int, array{type: 'file_citation', text: string, file_citation: array{file_id: string, quote: string}, start_index: int, end_index: int}|array{type: 'file_path', text: string, file_path: array{file_id: string}, start_index: int, end_index: int}>}}>, file_ids: array<int, string>}}  $attributes
+     */
+    public static function from(array $attributes): self
+    {
+        return new self(
+            $attributes['id'],
+            $attributes['object'],
+            ThreadMessageDeltaObject::from($attributes['delta'])
+        );
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function toArray(): array
```

Comment:
The current code for toArray throws the below error,
```json
[{"Illuminate\\Http\\JsonResponse":"HTTP/1.1 500 Internal Server Error\r\nCache-Control:         no-cache, private\r\nContent-Type:          application/json\r\nDate:                  Tue, 16 Apr 2024 14:04:12 GMT\r\nX-Ratelimit-Limit:     3600\r\nX-Ratelimit-Remaining: 3600\r\n\r\n{\"status\":\"ERROR\",\"message\":\"Undefined property: OpenAI\\\\Responses\\\\Threads\\\\Messages\\\\Delta\\\\ThreadMessageDeltaResponse::$role\"}"}] 
```
because the `role` variable does not exist in this level of the Open AI response, so replacing the current code of the  `toArray()` method with below will fix the issue:
```php
    public function toArray(): array
    {
        return [
            'id' => $this->id,
            'object' => $this->object,
            'delta' => $this->delta->toArray(),
        ];
    }
```
[Reference to the message delta response structure](https://platform.openai.com/docs/api-reference/assistants-streaming/message-delta-object)

Reply:
Complete oversight on my part, I changed both the `_construct` and `from` methods but forgot to change the `toArray`.
Many thanks for pointing this out, and resolving it.